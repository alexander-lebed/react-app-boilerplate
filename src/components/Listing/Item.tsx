import React from 'react';

interface Props {
    title: string,
    label: string
}

const Item = ({title, label}: Props) => {
    return (
        <div className="Item">
            <div><i className='fas fa-film' /> {title}</div>
            <div>{label}</div>
        </div>
    );
};

export default Item;