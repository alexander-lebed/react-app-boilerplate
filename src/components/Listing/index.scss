.Item {
  display: flex;
  justify-content: space-between;
  padding: 20px;
  border-top: 1px solid grey;
}