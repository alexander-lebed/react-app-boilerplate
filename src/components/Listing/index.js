// @flow
import React from 'react';
import Item from './Item';
import type {ListItem} from '../../types';

type Props = {
    list: Array<ListItem>
}

const Listing = (props: Props) => {

    // todo: add Create, Update, Remove functionality

    return (
        <div>
            <h1>Listing</h1>
            {props.list.map(item => (
                <Item key={item.id} title={item.title} label={item.label} />
            ))}
        </div>
    );
};

export default Listing;