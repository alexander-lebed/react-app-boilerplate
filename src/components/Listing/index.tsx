import React from 'react';
import Item from './Item';
import {ListItem} from '../../types';

interface Props {
    list: Array<ListItem>
}

const Listing = (props: Props) => {
    return (
        <div>
            <h1>Listing</h1>
            {props.list.map(item => (
                <Item key={item.id} title={item.title} label={item.label} />
            ))}
        </div>
    );
};

export default Listing;