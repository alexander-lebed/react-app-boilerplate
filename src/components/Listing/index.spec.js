// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme'
import Listing from './index';
import Item from './Item';

describe('Listing', () => {
    const props = {
        list: JSON.parse(dataString)
    };

    test('should match snapshot', () => {
        const wrapper = renderer.create(<Listing {...props} />);
        const tree = wrapper.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('should render list', () => {
        const wrapper = mount(<Listing {...props} />);
        const itemComp = wrapper.find(Item);
        expect(itemComp.exists()).toEqual(true);
        expect(itemComp.length).toEqual(3);
    })
});

const dataString = `
[
  {
    "id": "j54654jk4k",
    "title": "MovieName A",
    "label": "4,99 €"
  },
  {
    "id": "bg45tgeafn",
    "title": "MovieName B",
    "label": "3,99 €"
  },
  {
    "id": "456il45tf3",
    "title": "MovieName C",
    "label": "2,99 €"
  }
]
`;
