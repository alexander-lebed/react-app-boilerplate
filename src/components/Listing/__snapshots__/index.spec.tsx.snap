// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing should match snapshot 1`] = `
<div>
  <h1>
    Listing
  </h1>
  <div
    className="Item"
  >
    <div>
      <i
        className="fas fa-film"
      />
       
      MovieName A
    </div>
    <div>
      4,99 €
    </div>
  </div>
  <div
    className="Item"
  >
    <div>
      <i
        className="fas fa-film"
      />
       
      MovieName B
    </div>
    <div>
      3,99 €
    </div>
  </div>
  <div
    className="Item"
  >
    <div>
      <i
        className="fas fa-film"
      />
       
      MovieName C
    </div>
    <div>
      2,99 €
    </div>
  </div>
</div>
`;
